### DISCUSSION TITLE
LSP call for importing type annotations

### DISCUSSION THREAD

**Discussion by MarkBekooy** [microsoft/pyright#6120] [Q&A]
Say, I have the following very basic example function:
```python
def append(elem: int, lst):
    lst.append(elem)
    return lst
```

Programmatically, I add the type annotation `List[int]` to the source code and thus it becomes:
```python
def append(elem: int, lst: List[int]):
    lst.append(elem)
    return lst
```

Now I was wondering, does Pyright support an LSP call that can import the `List` type annotation and thus in essence add `from typing import List` to the top of the file that would contain this example function?

Without the import, Pyright otherwise mentions `"List" is not defined`, which I understand, but would like to solve.

**Comment by erictraut** [comment]
> does Pyright support an LSP call that can import the `List` type annotation...

No, pyright doesn't have any such functionality. I don't know how such a feature could be implemented. A type checker can't generally know what import statement must be added for an arbitrary symbol. Pyright does provide completion suggestions that include auto-import actions, but it's up to the user to decide which completion suggestion is appropriate. Doing this automatically, without any user intervention, would be very error prone.

In the specific case of `List[int]`, you could avoid the problem by using `list[int]` instead, since `list` is a builtin symbol.
